tests/nightly/test_large_array.py

```diff
@@ -217,6 +255,33 @@ def numpy_space_to_depth(x, blocksize):
     output = mx.nd.space_to_depth(data, 2)
     assert_almost_equal(output.asnumpy(), expected, atol=1e-3, rtol=1e-3)
 
+
+def test_diag():
+    h = np.random.randint(2,9)
+    w = np.random.randint(2,9)
+    a_np = np.random.random((LARGE_X, 64)).astype(np.float32)
```

Comment:
Why do you need to convert to np.float32 explicitly here?

Reply:
@apeforest by default `np.random.random((LARGE_X, 64))` makes it float64 by default.

>>> x = np.random.random((50000000, 64))
>>> x
array([[0.77259927, 0.17861939, 0.73446367, ..., 0.61712618, 0.00411382,
        0.80232583],
       [0.99131563, 0.29269588, 0.34078989, ..., 0.14693316, 0.68376766,
        0.13507782],
       [0.26176515, 0.54208053, 0.42594753, ..., 0.76032471, 0.30179728,
        0.83745653],
       ...,
       [0.45731445, 0.84679834, 0.02738181, ..., 0.27567623, 0.94721731,
        0.77850831],
       [0.65551258, 0.92503922, 0.43253718, ..., 0.23874736, 0.55155928,
        0.33177961],
       [0.95565209, 0.86454407, 0.75257631, ..., 0.31738383, 0.90281116,
        0.93569039]])
>>> x.dtype
dtype('float64')

 I am trying to keep it consistent with: https://github.com/apache/incubator-mxnet/blob/f680255fbff25818ed3eee0d4541d80b3c7b9d9d/tests/python/unittest/test_operator.py#L8029
So, I think float32 is required here.

Reply:
ok. mx.nd.array(a_np) is float32 by default. I will remove `astype('float32')` from next line